More flexibility in cartesian pattern generator

GiudGiud opened the pull request:
@joshua-smandych with this you dont need to use a pin generator for water or graphite
refs #30930

GiudGiud committed:
Add verbose stitiching option in patterned cartesianMG
refs #30930

GiudGiud committed:
Let user pick the stitching boundary
refs #30930

GiudGiud committed:
Add test for changing stitching boundary
refs #30930

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3006651/), step Generate coverage on b1cf7ca wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

### Reactor

<table>
  <thead>
    <tr>
      <th></th>
      <th>590e82</th>
      <th colspan=4>#30931 b1cf7c</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>93.75%</td>
      <td>93.76%</td>
      <td>+0.01%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>7910</td>
      <td>7921</td>
      <td>+11</td>
      <td>20</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>527</td>
      <td>527</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/30931/coverage_diff/reactor)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/30931/coverage/reactor)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/30931/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud committed:
Allow for a different ID for the same name of the stitching boundary

miaoyinb commented on modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C:
```diff
@@ -732,14 +743,19 @@ PatternedCartesianMeshGenerator::generate()
 
+            const auto stitching_boundary_id_base =
+                MooseMeshUtils::getBoundaryID(_stitching_boundary_name, *out_mesh);
+            const auto stitching_boundary_id_periph =
+                MooseMeshUtils::getBoundaryID(_stitching_boundary_name, *out_mesh);
```
```suggestion
            const auto stitching_boundary_id_periph =
                MooseMeshUtils::getBoundaryID(_stitching_boundary_name, *tmp_peripheral_mesh);
```

GiudGiud replied:
good catch!

miaoyinb commented on modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C:
```diff
@@ -792,15 +812,19 @@ PatternedCartesianMeshGenerator::generate()
   }
 
-  // Check if OUTER_SIDESET_ID is really the external boundary. Correct if needed.
+  // Check if stitching_boundary_id is really the external boundary. Correct if needed.
```
Previous, we are sure that `OUTER_SIDESET_ID` should be the external boundary. So this step is correction. However, for a user specified boundary id, I am not sure if we want to modify it. We should describe this behavior in doc.

GiudGiud replied:
added to the doco

miaoyinb commented on modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C:
```diff
@@ -959,14 +983,16 @@ PatternedCartesianMeshGenerator::generate()
   // Assign customized outer surface boundary id and name
   if (_external_boundary_id > 0)
-    MooseMesh::changeBoundaryId(*out_mesh, OUTER_SIDESET_ID, _external_boundary_id, false);
+    MooseMesh::changeBoundaryId(*out_mesh, stitching_boundary_id, _external_boundary_id, false);
```
Same, this only makes sense if `stitching_boundary_id` is the external boundary (and if not we correct it as above)

GiudGiud replied:
added to doco

miaoyinb reviewed (changes requested):
It generally looks good. Just want to make sure we document the code behavior when users provide a boundary that is not purely external.

Also, the hexagonal counterpart should also be updated for consistence.

GiudGiud committed:
Address review:
- document renaming behavior for external boundary ID in MG doco
- fix mistake in finding stitiching boundary id in periph mesh

miaoyinb commented:
This one looks good now. Do you want to update `PatternedHexMeshGenerator` accordingly here too for consistence?

I do not have a strong opinion on this, let me know your preference.

GiudGiud commented:
I dont have a user for that for the hex pattern. Imo it's a simple enough thing that we could port it over if the need arises
